Assets/Scripts/PlayerController.cs

```diff
@@ -79,11 +88,87 @@ void GetGroundChecker()
         }
     }
 
+
     void CheckBottomBoundary()
     {
         if (groundChecker.IsTouchingLayers(LayerMask.GetMask("BottomBoundary"))) 
         {
             SceneManager.LoadScene(SceneManager.GetActiveScene().buildIndex);
         }
     }
+    #endregion
+
+    #region Size Shifting
+
+    void OnTitan(InputValue value)
+    {
+        Debug.Log("Titan");
+
+        titan.SetActive(true);
+
+        pixie.SetActive(false);
+        nomad.SetActive(false);
+
+        maxJumpCount = 0;
+    }
+
+    void OnNomad(InputValue value)
+    {
+        Debug.Log("Nomad");
+
+        nomad.SetActive(true);
+
+        pixie.SetActive(false);
+        titan.SetActive(false);
+
+        maxJumpCount = 1;
+    }
+
+    void OnPixie(InputValue value)
+    {
+        Debug.Log("Pixie");
+
+        pixie.SetActive(true);
+
+        titan.SetActive(false);
+        nomad.SetActive(false);
+
+        maxJumpCount = 2;
+    }
+
+    void GetCharacters()
+    {
+        Transform pixie = transform.Find("Pixie");
+        Transform nomad = transform.Find("Nomad");
+        Transform titan = transform.Find("Titan");
+
+        if (pixie == null)
+        {
+            Debug.LogError("Pixie mesh not found!");
+        }
+        else
+        {
+            this.pixie = pixie.gameObject;
+        }
+
+        if (nomad == null)
+        {
+            Debug.LogError("Nomad mesh not found!");
+        }
+        else
+        {
+            this.nomad = nomad.gameObject;         
+        }
+
+        if (titan == null)
+        {
+            Debug.LogError("Titan mesh not found!");
+        }
+        else
+        {
```

Comment:
Great job here with the logic, just a tip, for if/else repetitive statements you can use `switch` cases instead, here's a cleaner version:

```
void GetCharacters()
{
    string[] forms = { "Pixie", "Nomad", "Titan" };

    foreach (string form in forms)
    {
        Transform currentForm = transform.Find(form);

        if (currentForm == null)
        {
            Debug.LogError($"{form} mesh not found!");
        }
        else
        {
            switch (form)
            {
                case "Pixie":
                    this.pixie = currentForm.gameObject;
                    break;
                case "Nomad":
                    this.nomad = currentForm.gameObject;
                    break;
                case "Titan":
                    this.titan = currentForm.gameObject;
                    break;
            }
        }
    }
}
```

Feel free to use the code above if you'd like

Reply:
Thank you :) Done.